app/Database/CourseQueries.hs

```diff
@@ -202,16 +202,18 @@ getGraph graphName =
                 return (Just response) :: SqlPersistM (Maybe Response)
 
 -- | Retrieves the prerequisites for a course (code) as a string.
-prereqsForCourse :: T.Text -> IO (Either String T.Text)
+-- Also retrieves the actual course code in the database in case
+-- the one the user inputs doesn't match it exactly
+prereqsForCourse :: T.Text -> IO (Either String (T.Text, T.Text))
 prereqsForCourse courseCode = runSqlite databasePath $ do
-    course <- selectFirst [CoursesCode ==. courseCode] []
+    course <- selectFirst [CoursesCode <-. [T.toUpper courseCode, T.toUpper courseCode `T.append` "H1", courseCode `T.append` "Y1"]] []
```

Comment:
Use a local variable to store `T.toUpper courseCode`. Note that you missed the `T.toUpper` in the last element, which is better avoided with the local variable.